pom.xml

```diff
@@ -2590,6 +2590,25 @@
                 <extensions>true</extensions>
             </plugin>
 
+            <plugin>
+                <groupId>org.gaul</groupId>
+                <artifactId>modernizer-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>modernizer-production-code</id>
+                        <goals>
+                            <goal>modernizer</goal>
+                        </goals>
+                        <configuration>
+                            <violationsFiles>
+                                <violationsFile>${air.main.basedir}/.mvn/modernizer/violations-production-code-only.xml</violationsFile>
```

Comment:
Not sure I like the idea of rules that apply only to one part of the codebase.

Reply:
It gets murkier for modules that contain code that’s only used for tests in other modules (e.g., common testing infrastructure)

Reply:
Agreed! I am definitely not a fan of such rules either. I am not sure who will want to spend time on giving names to all the executors in tests.

Reply:
Maybe we should just do it once and move on?

Reply:
agreed, this is a good idea. just that i probably won't have time to do this myself.

Reply:
So let's merge this as-is and I'll follow up with the rest of the thread pools, wdyt?

Reply:
works for me